src/items/coherence.md

```diff
@@ -0,0 +1,79 @@
+## What is coherence and why do we care?
```

Comment:
I think this text should mainly explain that overlap is prevented (and the rules for that) and that orphans are not allowed (and the rules for that).

Question to @nikomatsakis: is the orphan rule actually part of preventing overlap in the compiler?